### ISSUE TITLE
zautomatyzowana autoryzacja ?

### ISSUE THREAD

**Issue by tony6636** [allegro/allegro-api#626]
Cześć,
czy jest sposób aby zautomatyzować autoryzację użytkownika https://developer.allegro.pl/auth/#user ? Chciałbym użyć API w swoim projekcie, a autoryzowanie usera w przeglądarce co 10s będzie problemem.

**Comment by collector1871** [comment]
<reklama>
jakbyś coś miał fajnego w tym zakresie, jakąś ciekawą funkcjonalność, to zapraszam:
[https://github.com/collector1871/allsi-login](https://github.com/collector1871/allsi-login)
</reklama>

**Comment by mhaligowski** [comment]
Czy to oznacza, że nie ma możliwości autoryzacji typu serwer-serwer? Znacząco to utrudnia sprawę przy automatyzacji własnego konta, a przechowywanie tokenów jest dość kłopotliwe i naraża moją aplikację na poważne niebezpieczeństwa. Czy nie ma planów dodania takich autoryzacji, tym bardziej że client id oraz client secret są już generowane? 

Dodam, że jest to zwykły flow zdefiniowany w ramach standardu RFC: https://tools.ietf.org/html/rfc6749#section-4.4

**Comment by PawelTaberski** [comment]
Sposobem jaki opisałeś serwer-serwer nie uzyskasz tokena usera, jednakże możesz korzystać wtedy z ogólnie dostępnych metod np listing. Niedługo udostępnimy "device flow" czyli wsparcie dla aplikacji headless. Jednakże i tak jeżeli aplikacja ma działać w imieniu usera to musi przechowywać tokeny.

**Comment by mhaligowski** [comment]
@PawelTaberski niestety nie rozwiązuje to w żaden sposób problemu w jaki sposób uwierzytelnić moją aplikację do dostępu do zasobów w moim koncie moimi danymi. Czy są jakiekolwiek plany odnośnie zgodności ze standardem OAuth 2.0?

**Comment by PawelTaberski** [comment]
Tak, jak Panu pisałem jesteśmy zgodni z OAuth2.0 i aplikacja może użyć client credentials, żeby otrzymać token autoryzacyjny. Jednakże jak wspominałem nie ma dostępu wtedy do danych użytkownika i może korzystać tylko z takich zasobów jak np. listing.  Bo autoryzuje Pan wtedy aplikację nie użytkownika. Planujemy wdrożenie specjalnego flow dla aplikacji typu headless, obecnie innych zmian nie planujemy.

**Comment by mhaligowski** [comment]
OK, ma sens.

Czy istnieje możliwość wygenerowania tokena na dłużej niż 12 godzin? ew. czy aplikacje headless pozwolą na tokeny na dłuższy okres czasu? Kiedy się spodziewać tych zmian?

**Comment by mhaligowski** [comment]
I jeszcze jedno pytanie: czy refresh token również przekierowuje na adres podany w `redirect_uri`?

**Comment by PawelTaberski** [comment]
Nie planujemy wydłużenia czasu życia tokenów. Odnośnie refresh tokena komplet informacji znajdzie Pan [tutaj](https://developer.allegro.pl/auth/#refresh-token). Co do wprowadzenia zmian nie znam dokładnej daty, zakładam że jest to kwestia najbliższych tygodni.
